homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,382 @@
+"""
+Support for Cambridge Audio Network Audio Players (StreamMagic platform).
+
+Example configuration.yaml entry:
+media_player:
+    platform: cambridgeaudio
+    host: 192.168.x.y
+    command_off: "OFF"
+    name: "CA 851N"
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.cambridgeaudio/
+"""
+import logging
+# pylint: disable=unused-import
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_PAUSE, SUPPORT_PLAY, SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    SUPPORT_STOP, SUPPORT_SEEK, SUPPORT_SHUFFLE_SET,
+    SUPPORT_SELECT_SOURCE, SUPPORT_CLEAR_PLAYLIST,
+    SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_MUSIC)
+
+from homeassistant.const import (STATE_ON, STATE_OFF, STATE_UNKNOWN,
+                                 STATE_PLAYING, STATE_PAUSED, STATE_IDLE,
+                                 CONF_HOST, CONF_NAME, CONF_COMMAND_OFF)
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['stream_magic==0.16']
+DOMAIN = 'cambridgeaudio'
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Cambridge Audio Streamer'
+CONF_SOURCES = 'sources'
+KNOWN_HOSTS = []
```

Comment:
This should not be a global list but stored in `hass.data`, eg `hass.data['cambridgeaudio_hosts']`.